### ISSUE TITLE
Using `style` HUD flag leads into all black text with old server

### ISSUE THREAD

**Issue by AFCMS** [minetest/minetest#11539]
##### Minetest version

```
Minetest 5.5.0-dev-b3b075e (Linux)
Using Irrlicht 1.9.0mt3
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=0
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="."

```

##### Summary
Using #11478 `style` HUD flag leads into all black text with old clients. 

##### Steps to reproduce
Execute `/title test` with this mcl2 pr in both `5.5-dev` and `5.4.1`: https://git.minetest.land/MineClone2/MineClone2/pulls/1778

Minetest 5.5-dev:
![image](https://user-images.githubusercontent.com/61794590/129564089-8da3e366-cec2-4b45-9e83-82c38e3f2b19.png)

Minetest 5.4.1:
![image](https://user-images.githubusercontent.com/61794590/129564333-bda40afa-7c7a-43b8-8d28-48e921f0097d.png)

**Comment by appgurueu** [comment]
Are you using hud_change?

**Comment by AFCMS** [comment]
> Ideally, the engine should IMO throw an error instead of just updating the number field as a fallback. If you want to hotfix this, just don't send the hud_change if your server is <= 5.4.

:+1:

**Comment by AFCMS** [comment]
@appgurueu Isn't adding a feature flag a better idea?

**Comment by sfan5** [comment]
We can't retroactively fix this for older servers but 5.5.0 should be made to ignore unknown enum values so this doesn't happen in the future.
